impl/src/test/java/org/ehcache/docs/GettingStarted.java

```diff
@@ -245,15 +232,14 @@ public void writeThroughCache() throws ClassNotFoundException {
   
   @Test
   public void writeBehindCache() throws ClassNotFoundException {
-    
-    // tag::writeBehindCache[]    
+    // tag::writeBehindCache[]
     CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder().build(true);
     
     Class<CacheLoaderWriter<?, ?>> klazz = (Class<CacheLoaderWriter<?, ?>>) (Class) (SampleLoaderWriter.class);
     
     final Cache<Long, String> writeBehindCache = cacheManager.createCache("writeBehindCache",
         CacheConfigurationBuilder.newCacheConfigurationBuilder()
-            .add(new DefaultCacheLoaderWriterConfiguration(klazz, singletonMap(41L, "zero"))) // <1>
+            .withLoaderWriter((SampleLoaderWriter) new SampleLoaderWriter<Long, String>(singletonMap(41L, "zero"))) // <1>
```

Comment:
I agree, but the #704 changes will make the compiler able to infer the types and make those casts useless.

My idea was to leave them as such as this is still correct code, and the #704 rework will make the ugliness go away.